_includes/manuals/2.4/3.6_upgrade.md

```diff
@@ -31,113 +52,138 @@ For more information about how to backup your instance head over to
 
-##### Step 2 (B) - Debian or Ubuntu package (deb) and installer setups
+<div class="upgrade_os upgrade_os_el8">
+To upgrade an existing Foreman installation, first update with the
+appropriate foreman-release package:
 
-Upgrading from the last release to {{page.version}} has been tested on both
-Debian and Ubuntu. Updating the packages will upgrade the application and
-automatically migrate the database.
+{% highlight bash %}
+dnf upgrade https://yum.theforeman.org/releases/{{page.version}}/el8/x86_64/foreman-release.rpm
+{% endhighlight %}
 
-First edit `/etc/apt/sources.list.d/foreman.list` and change the release
-number from the previous release or "stable" to `{{ page.version }}`:
+Clean up the yum metadata cache:
 
-    deb http://deb.theforeman.org/ buster {{ page.version }}
-    deb http://deb.theforeman.org/ plugins {{ page.version }}
+{% highlight bash %}
+dnf clean metadata
+{% endhighlight %}
 
 Next upgrade all Foreman packages:
 
-    apt-get update
-    apt-get --only-upgrade install ruby\* foreman\*
-
-You can skip to Step 3.
-
-##### Step 2 (C) - Downloaded release (tar.bz2)
-
-- Uncompress the new program archive in a *new directory*.
-- Copy your database settings-file `config/database.yml` into the new `config` directory.
-
-VERY IMPORTANT: do NOT overwrite `config/settings.yml` with the old one.
-
-##### Step 2 (D) - git checkouts only
+{% highlight bash %}
+dnf upgrade ruby\* foreman\*
+{% endhighlight %}
 
-*Please note* that there can be significant differences between stable and
-development branches.  You MUST take care to select a branch and make sure you
-get the correct one.
+Foreman runs best on PostgreSQL 12 but the default version on EL8 is 10. It is recommended to upgrade by [switching module streams](https://docs.fedoraproject.org/en-US/modularity/using-modules-switching-streams/).
 
-1. Go to the Foreman root directory and run the following command:
+First check which version of PostgreSQL is currently installed:
 
-For staying on the stable branch:
+{% highlight bash %}
+rpm -qv postgresql-server
+{% endhighlight %}
 
-- git checkout {{page.version}}-stable
-- git pull
+If you're already on 12 or don't have the PostgreSQL server installed, you can continue with step 3. Otherwise upgrade:
 
-There may have been changes to the ruby gem or node module versions, to update
-run the following commands:
+{% highlight bash %}
+systemctl stop postgresql.service
+# https://bugzilla.redhat.com/1935301
+sed -i '/^data_directory =/d' /var/lib/pgsql/data/postgresql.conf
+dnf module reset postgresql
+dnf module install postgresql:12
+dnf install postgresql-upgrade
+postgresql-setup --upgrade
+systemctl start postgresql.service
+{% endhighlight %}
+</div>
 
-    bundle update
-    npm update
+<div class="upgrade_os upgrade_os_debian10 upgrade_os_ubuntu1804">
+Upgrading from the last release to {{page.version}} has been tested. Updating
+the packages will upgrade the application and automatically migrate the
+database.
```

Comment:
do deb packages still run migrations in postinstall?

Reply:
Yes, https://github.com/theforeman/foreman-packaging/blob/9fe8df5457c502b9b48bc38cebedebfada41486f/debian/buster/foreman/foreman-postgresql.postinst#L35-L36 (and other `postinst` files)